presto-main/src/test/java/com/facebook/presto/block/TestMapBlock.java

```diff
@@ -0,0 +1,135 @@
+            }
+            result[i] = builder.build();
+        }
+        return result;
+    }
+
+    private void testWith(Map<String, Long>[] expectedValues)
+    {
+        BlockBuilder blockBuilder = createBlockBuilderWithValues(expectedValues);
+
+        assertBlock(blockBuilder, expectedValues);
+        assertBlock(blockBuilder.build(), expectedValues);
+        assertBlockFilteredPositions(expectedValues, blockBuilder.build(), Ints.asList(0, 1, 3, 4, 7));
+        assertBlockFilteredPositions(expectedValues, blockBuilder.build(), Ints.asList(2, 3, 5, 6));
+
+        Map<String, Long>[] expectedValuesWithNull = (Map<String, Long>[]) alternatingNullValues(expectedValues);
+        BlockBuilder blockBuilderWithNull = createBlockBuilderWithValues(expectedValuesWithNull);
+
+        assertBlock(blockBuilderWithNull, expectedValuesWithNull);
+        assertBlock(blockBuilderWithNull.build(), expectedValuesWithNull);
+        assertBlockFilteredPositions(expectedValuesWithNull, blockBuilderWithNull.build(), Ints.asList(0, 1, 5, 6, 7, 10, 11, 12, 15));
+        assertBlockFilteredPositions(expectedValuesWithNull, blockBuilderWithNull.build(), Ints.asList(2, 3, 4, 9, 13, 14));
+    }
+
+    private BlockBuilder createBlockBuilderWithValues(Map<String, Long>[] maps)
+    {
+        MapType mapType = mapType(VARCHAR, BIGINT);
+        BlockBuilder mapBlockBuilder = mapType.createBlockBuilder(new BlockBuilderStatus(), 1);
+        for (Map<String, Long> map : maps) {
+            createBlockBuilderWithValues(map, mapBlockBuilder);
+        }
+        return mapBlockBuilder;
+    }
+
+    private void createBlockBuilderWithValues(Map<String, Long> map, BlockBuilder mapBlockBuilder)
+    {
+        if (map == null) {
+            mapBlockBuilder.appendNull();
+        }
+        else {
+            BlockBuilder elementBlockBuilder = mapBlockBuilder.beginBlockEntry();
+            for (Map.Entry<String, Long> entry : map.entrySet()) {
+                VARCHAR.writeSlice(elementBlockBuilder, utf8Slice(entry.getKey()));
+                BIGINT.writeLong(elementBlockBuilder, entry.getValue());
+            }
+            mapBlockBuilder.closeEntry();
+        }
+    }
+
+    @Override
+    protected <T> void assertPositionValue(Block block, int position, T expectedValue)
+    {
+        if (expectedValue instanceof Map) {
+            assertValue(block, position, (Map<String, Long>) expectedValue);
+            return;
+        }
+        super.assertPositionValue(block, position, expectedValue);
+    }
+
+    private void assertValue(Block mapBlock, int position, Map<String, Long> map)
+    {
+        MapType mapType = mapType(VARCHAR, BIGINT);
+
+        if (map == null) {
+            assertTrue(mapBlock.isNull(position));
+        }
+        else {
+            assertFalse(mapBlock.isNull(position));
+            MapElementBlock elementBlock = (MapElementBlock) mapType.getObject(mapBlock, position);
```

Comment:
Actually, this is not ok. During transition period, maps can still be `ArrayBlock<InterleavedBlock>` instead of `MapBlock`.